### PR TITLE
Applying Redux(toolkit) and Filter Feature(movie-cards)

### PR DESCRIPTION
**Pull request by rochagbriel**
- Installed Redux and Redux Toolkit;
- Created the reducer slices;
- Created and set the store with the reducer slices;
- Modified components so they read from the store and dispatch actions to the store;
- Added the ability to filter out the list of movie-cards.

### CODE REVIEW

File: src/components/main-view/main-view.jsx [hunk 1]
```diff
@@ -8,16 +8,19 @@ import { ProtectedRoutes } from './protected-routes';
...
-  const storedUser = JSON.parse(localStorage.getItem('user'));
-  const storedToken = localStorage.getItem('token');
-  const [user, setUser] = useState(storedUser ? storedUser : null);
-  const [token, setToken] = useState(storedToken ? storedToken : null);
-  const [movies, setMovies] = useState([]);
+  const dispatch = useDispatch();
+
+  const user = useSelector((state) => state.user.user);
+  const token = useSelector((state) => state.user.token);
+  const movies = useSelector((state) => state.movies.list);
```
Comment (Stanley-Okwii) [inline src/components/main-view/main-view.jsx:20]: 🤩 Great use of useSelector hook.

File: src/components/main-view/main-view.jsx [hunk 2]
```diff
@@ -115,50 +100,20 @@ export const MainView = () => {
...
-                        md={6}
-                        xs={12}
-                      >
-                        <MovieCard movie={movie} />
-                      </Col>
-                    ))}
-                  </>
-                )}
+              <ProtectedRoutes>
+                        <MoviesList />
```
Comment (Stanley-Okwii) [inline src/components/main-view/main-view.jsx:104]: Well done 💪🏻, refactoring this...

File: src/components/main-view/protected-routes.jsx [hunk 3]
```diff
@@ -1,10 +1,11 @@
 import { Navigate, Outlet } from 'react-router-dom';
+import { useSelector } from 'react-redux';
 
 export const ProtectedRoutes = ({
-  user,
   children,
   redirectPath = '/login',
 }) => {
+  const user = useSelector((state) => state.user.user);
```
Comment (Stanley-Okwii) [inline src/components/main-view/protected-routes.jsx:8]: 👌🏻

File: src/components/movie-view/movie-view.jsx [hunk 4]
```diff
@@ -107,7 +111,7 @@ export const MovieView = ({ movies, user, token, updateUser }) => {
         )}
       </Col>
       <Container>
-        <SimilarMovies movies={movies} movie={movie} />
+        <SimilarMovies movie={movie} />
```
Comment (Stanley-Okwii) [inline src/components/movie-view/movie-view.jsx:114]: 👌🏻

File: src/components/navigation-bar/navigation-bar.jsx [hunk 5]
```diff
@@ -36,7 +41,7 @@ export const NavigationBar = ({ user, onLoggedOut }) => {
                 <Nav.Link className='link-header-light' as={Link} to='/profile'>
                   Profile
                 </Nav.Link>
-                <Nav.Link onClick={onLoggedOut} className='ms-lg-auto'>
+                <Nav.Link onClick={() => dispatch(setUser(null),setToken(null),localStorage.clear(),window.location.reload())} className='ms-lg-auto'>
```
Comment (Stanley-Okwii) [inline src/components/navigation-bar/navigation-bar.jsx]: ⛏️If you can always extract this to a single function to simplify the complexity of this line.

File: src/index.jsx [hunk 6]
```diff
@@ -1,15 +1,19 @@
...
 
 import './index.scss';
 
 // Main component (will eventually use all the others)
 const MyFlixApplication = () => {
   return (
-    <Container>
-      <MainView />
-    </Container>
+    <Provider store={store}>
```
Comment (Stanley-Okwii) [inline src/index.jsx:12]: Nice 🥳

**Review by Stanley-Okwii — APPROVED** [review]
I must commend you on the outstanding refactoring work you've accomplished. It brings me great satisfaction to observe the multitude of changes you've implemented in this pull request. Your dedication and attention to detail truly shine through. 🎉

File: src/components/main-view/main-view.jsx (hunk 1 above)
Comment (rochagbriel) [reply src/components/main-view/main-view.jsx:20]: thank you!

File: src/components/navigation-bar/navigation-bar.jsx (hunk 5 above)
Comment (rochagbriel) [reply src/components/navigation-bar/navigation-bar.jsx]: done! thanks

**Pull request merged**